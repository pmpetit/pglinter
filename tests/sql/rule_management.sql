-- Test for dblinter rule management functionality
BEGIN;

DROP EXTENSION IF EXISTS dblinter CASCADE;

CREATE TABLE IF NOT EXISTS test_table_for_rules (
    id INT,
    name TEXT
);

CREATE EXTENSION IF NOT EXISTS dblinter;

-- Show initial rule status
SELECT dblinter.show_rules();

-- Test enabling and disabling a specific rule
SELECT dblinter.is_rule_enabled('B001') AS b001_initially_enabled;

-- Disable B001 rule
SELECT dblinter.disable_rule('B001') AS b001_disabled;

-- Check if it's disabled
SELECT dblinter.is_rule_enabled('B001') AS b001_after_disable;

-- Run base check (should skip B001)
SELECT dblinter.perform_base_check();

-- Re-enable B001 rule
SELECT dblinter.enable_rule('B001') AS b001_enabled;

-- Check if it's enabled again
SELECT dblinter.is_rule_enabled('B001') AS b001_after_enable;

-- Test with non-existent rule
SELECT dblinter.disable_rule('NONEXISTENT') AS nonexistent_disable;

-- Show final rule status
SELECT dblinter.show_rules();

ROLLBACK;
