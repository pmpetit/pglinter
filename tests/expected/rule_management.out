-- Test for pglinter rule management functionality
CREATE EXTENSION pglinter;
CREATE TABLE test_table_for_rules (
    id INT,
    name TEXT
);
-- Show initial rule status
SELECT pglinter.show_rules();
NOTICE:  üìã pglinter Rule Status:
NOTICE:  ============================================================
NOTICE:  Code   Status   Name                                    
NOTICE:  ------------------------------------------------------------
NOTICE:  B001   ‚úÖ ON     HowManyTableWithoutPrimaryKey           
NOTICE:  B002   ‚úÖ ON     HowManyRedudantIndex                    
NOTICE:  B003   ‚úÖ ON     HowManyTableWithoutIndexOnFk            
NOTICE:  B004   ‚úÖ ON     HowManyUnusedIndex                      
NOTICE:  B005   ‚úÖ ON     UnsecuredPublicSchema                   
NOTICE:  B006   ‚úÖ ON     HowManyObjectsWithUppercase             
NOTICE:  B007   ‚úÖ ON     HowManyTablesNeverSelected              
NOTICE:  B008   ‚úÖ ON     HowManyTablesWithFkOutsideSchema        
NOTICE:  B015   ‚úÖ ON     HowManyTableSharingSameTrigger          
NOTICE:  C002   ‚úÖ ON     PgHbaEntriesWithMethodTrustOrPasswordShouldNotExists
NOTICE:  C003   ‚úÖ ON     PasswordEncryptionIsMd5                 
NOTICE:  S001   ‚úÖ ON     SchemaWithDefaultRoleNotGranted         
NOTICE:  S002   ‚úÖ ON     SchemaPrefixedOrSuffixedWithEnvt        
NOTICE:  T001   ‚úÖ ON     TableWithoutPrimaryKey                  
NOTICE:  T002   ‚úÖ ON     TableWithRedundantIndex                 
NOTICE:  T003   ‚úÖ ON     TableWithFkNotIndexed                   
NOTICE:  T004   ‚úÖ ON     TableWithPotentialMissingIdx            
NOTICE:  T005   ‚úÖ ON     TableWithFkOutsideSchema                
NOTICE:  T006   ‚úÖ ON     TableWithUnusedIndex                    
NOTICE:  T007   ‚úÖ ON     TableWithFkMismatch                     
NOTICE:  T008   ‚úÖ ON     TableWithRoleNotGranted                 
NOTICE:  T009   ‚úÖ ON     ReservedKeyWord                         
NOTICE:  T010   üî¥ OFF    TableWithSensibleColumn                 
NOTICE:  T015   ‚úÖ ON     TableSharingSameTrigger                 
NOTICE:  ============================================================
NOTICE:  üìä Summary: 23 enabled, 1 disabled
 show_rules 
------------
 t
(1 row)

-- Test enabling and disabling a specific rule
SELECT pglinter.is_rule_enabled('B001') AS b001_initially_enabled;
 b001_initially_enabled 
------------------------
 t
(1 row)

-- Disable B001 rule
SELECT pglinter.disable_rule('B001') AS b001_disabled;
NOTICE:  üî¥ Rule B001 has been disabled
 b001_disabled 
---------------
 t
(1 row)

-- Check if it's disabled
SELECT pglinter.is_rule_enabled('B001') AS b001_after_disable;
 b001_after_disable 
--------------------
 f
(1 row)

-- Run base check (should skip B001)
SELECT pglinter.perform_base_check();
WARNING:  pglinter base check failed: Failed to execute q2 rule B015: Database error: SpiTupleTable positioned before the start or after the end
 perform_base_check 
--------------------
 f
(1 row)

-- Re-enable B001 rule
SELECT pglinter.enable_rule('B001') AS b001_enabled;
NOTICE:  ‚úÖ Rule B001 has been enabled
 b001_enabled 
--------------
 t
(1 row)

-- Check if it's enabled again
SELECT pglinter.is_rule_enabled('B001') AS b001_after_enable;
 b001_after_enable 
-------------------
 t
(1 row)

-- Test with non-existent rule
SELECT pglinter.disable_rule('NONEXISTENT') AS nonexistent_disable;
WARNING:  ‚ö†Ô∏è  Rule NONEXISTENT not found
 nonexistent_disable 
---------------------
 f
(1 row)

-- Show final rule status
SELECT pglinter.show_rules();
NOTICE:  üìã pglinter Rule Status:
NOTICE:  ============================================================
NOTICE:  Code   Status   Name                                    
NOTICE:  ------------------------------------------------------------
NOTICE:  B001   ‚úÖ ON     HowManyTableWithoutPrimaryKey           
NOTICE:  B002   ‚úÖ ON     HowManyRedudantIndex                    
NOTICE:  B003   ‚úÖ ON     HowManyTableWithoutIndexOnFk            
NOTICE:  B004   ‚úÖ ON     HowManyUnusedIndex                      
NOTICE:  B005   ‚úÖ ON     UnsecuredPublicSchema                   
NOTICE:  B006   ‚úÖ ON     HowManyObjectsWithUppercase             
NOTICE:  B007   ‚úÖ ON     HowManyTablesNeverSelected              
NOTICE:  B008   ‚úÖ ON     HowManyTablesWithFkOutsideSchema        
NOTICE:  B015   ‚úÖ ON     HowManyTableSharingSameTrigger          
NOTICE:  C002   ‚úÖ ON     PgHbaEntriesWithMethodTrustOrPasswordShouldNotExists
NOTICE:  C003   ‚úÖ ON     PasswordEncryptionIsMd5                 
NOTICE:  S001   ‚úÖ ON     SchemaWithDefaultRoleNotGranted         
NOTICE:  S002   ‚úÖ ON     SchemaPrefixedOrSuffixedWithEnvt        
NOTICE:  T001   ‚úÖ ON     TableWithoutPrimaryKey                  
NOTICE:  T002   ‚úÖ ON     TableWithRedundantIndex                 
NOTICE:  T003   ‚úÖ ON     TableWithFkNotIndexed                   
NOTICE:  T004   ‚úÖ ON     TableWithPotentialMissingIdx            
NOTICE:  T005   ‚úÖ ON     TableWithFkOutsideSchema                
NOTICE:  T006   ‚úÖ ON     TableWithUnusedIndex                    
NOTICE:  T007   ‚úÖ ON     TableWithFkMismatch                     
NOTICE:  T008   ‚úÖ ON     TableWithRoleNotGranted                 
NOTICE:  T009   ‚úÖ ON     ReservedKeyWord                         
NOTICE:  T010   üî¥ OFF    TableWithSensibleColumn                 
NOTICE:  T015   ‚úÖ ON     TableSharingSameTrigger                 
NOTICE:  ============================================================
NOTICE:  üìä Summary: 23 enabled, 1 disabled
 show_rules 
------------
 t
(1 row)

DROP TABLE test_table_for_rules CASCADE;
DROP EXTENSION pglinter CASCADE;
